Caffeine/Caffeine/ViewControllers/LoginViewController.swift

```diff
@@ -70,23 +73,28 @@ extension LoginViewController {
     }
 
     @objc private func turnOnHomeScreen() {
-        if loginViewModel.checkForInvalidInut(self.emailField) == true {
-            self.showAlertButtonTapped(self.loginButton)
-            self.emailField.text = ""
-            self.passwordField.text = ""
+        if self.wrongInputPromt != nil { // Dismiss the view from here
+            self.wrongInputPromt.removeFromSuperview()
+            self.timerForPromt.invalidate()
         }
-        else {
+
+        if self.loginViewModel.checkForAllValidFields(self.emailField, self.passwordField) == true {
             //TODO: navigate to next screen
+            print("Yay")
+        }
+        else {
+            self.view.addSubview(self.wrongInputPromt)
+            self.activateWrongInputPromtLabelViewConstraints()
+            self.timerForPromt = initTimerForPromt()
         }
     }
 }
 
-//MARK: Checking for valid input
 extension LoginViewController {
-    private func showAlertButtonTapped(_ sender: UIButton) {
-        let alert = UIAlertController(title: "Klaida!", message: "Neteisingas El.paštas arba slaptažodis", preferredStyle: UIAlertController.Style.alert)
-        alert.addAction(UIAlertAction(title: "OK", style: UIAlertAction.Style.default, handler: nil))
-        self.present(alert, animated: true, completion: nil)
+    @objc private func dismissInvalidInputPromt() {
+        if self.wrongInputPromt != nil { // Dismiss the view from here
+            self.wrongInputPromt.removeFromSuperview()
+        }
```

Comment:
cia irgi: Comparing non-optional value of type 'UILabel' to 'nil' always returns true

Reply:
Tu sita uilabel susikures esi not nil. Ir nu techniskai neimanoma, kad jis butu nil